web/pandas_web.py

```diff
@@ -223,7 +224,24 @@ def home_add_releases(context):
         with open(pathlib.Path(context["target_path"]) / "releases.json", "w") as f:
             json.dump(releases, f, default=datetime.datetime.isoformat)
 
-        for release in releases:
+        parsed_releases = map(lambda r: (version.parse(r["tag_name"]), r), releases)
+
+        # A version is obsolete if it's not the latest minor version.
+        # In the list ["1.4.0", "1.4.1", "1.4.2", "1.5.0", "1.5.1"], versions
+        # "1.4.0", "1.4.1", and "1.5.0" are obsolete,
+        # We don't need 1.5.0 because 1.5.1 has the same features with some bugs fixed
+
+        # Filter obsolete versions
+        latest_releases = []
+        prev_version = version.Version("0.0.0")
+        for v, release in sorted(parsed_releases, reverse=True):
```

Comment:
I would prefer to not look twice, and instead do the filtering in the existing loop below.

Reply:
I agree that it's a better practice. Just combined the two together!